contrib/scripts/quarantine.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+set -e
+set -o pipefail
+
+indent() {
+    sed 's/^/  /'
+}
+
+main() {
+    tmpfile=$(mktemp)
+    trap 'rm -f -- $tmpfile' EXIT
+
+    if [ $# -lt 1 ] || [ $# -gt 1 ]; then
+        >&2 echo "usage: $0 <focus-phrase>"
+        return 1
+    fi
+
+    if ! git grep -q "$1"; then
+        >&2 echo "Unable to find phrase."
+        return 1
+    fi
+
+    if ! git diff --quiet || ! git diff --quiet --cached; then
+        >&2 echo "Local changes in the git tree break this script. Stage your changes to continue."
+        return 1
+    fi
+
+    files=( $(git grep -l "$1" test/*/*go) )
+    for f in "${files[@]}"; do
+        commits=( $(git blame $f | grep "$1" |  awk '{ print $1; }') )
+        authors=()
+        for c in "${commits[@]}"; do
+            authors+=( "$(git log -1 $c --pretty='%aN <%aE>')" )
+        done
```

Comment:
I'm not convinced this is going to be very relevant in general. Looking at the subsequent commits, it seems to work ~50% of the time. Wouldn't relying on the codeowners for the pull request be enough?

Reply:
The idea in principle is to try to find the original author of the test to try to provide flake failure feedback to them so that they can learn about when a test is written in a way that doesn't reliably fail, and hopefully figure out how to improve the tests they write over time.

In practice this little bit doesn't directly notify the author and sometimes the author of the line we're changing isn't the most relevant line, but I still think that this hint is a little more interesting as a datapoint about who was last involved in enabling the test, which we can consider using to reach out and have the discussion about the flaky test.